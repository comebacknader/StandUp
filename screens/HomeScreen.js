import React from 'react';

import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableHighlight,
} from 'react-native';

import { ScrollView } from 'react-native-gesture-handler';
import { Layout, Typography, Color } from '../styles';


const HomeScreen = ({ navigation }) => {

  const onPressImpressions = () => {
    navigation.navigate('Impressions')
  }

  const onPressBits = () => {
    navigation.navigate('Bits')
  }
  
  return (
      <View>
        <ScrollView>
          <TouchableHighlight activeOpacity={0.6}
            underlayColor={Color.LIGHTGRAY}
            onPress={onPressImpressions}>
            <View style={[Layout.flexAndDirectionRow, Layout.alignItemsCenter, Layout.paddingTen, Layout.bottomBorderGray]}>
              <Image source={require('../src/assets/images/theater.png')} style={[Layout.smallImage]} />
              <Text style={[Layout.marginLeft20, Typography.TitleFont]}>Impressions</Text>
            </View>
          </TouchableHighlight>
          <TouchableHighlight activeOpacity={0.6}
            underlayColor={Color.LIGHTGRAY}
            onPress={onPressBits}>
            <View style={[Layout.flexAndDirectionRow, Layout.alignItemsCenter, Layout.paddingTen, Layout.bottomBorderGray]}>
              <Image source={require('../src/assets/images/actor.png')} style={[Layout.smallImage]} />
              <Text style={[Layout.marginLeft20, Typography.TitleFont]}>Bits</Text>
            </View>
          </TouchableHighlight>
        </ScrollView>
      </View>
  )
}

const styles = StyleSheet.create({});

export default HomeScreen;