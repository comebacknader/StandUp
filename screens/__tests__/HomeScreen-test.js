/**
 * @format
 */

import 'react-native';
import React from 'react';
import HomeScreen from '../HomeScreen';

// Note: test renderer must be required after react-native.
import { create } from 'react-test-renderer';

it('renders the HomeScreen component', () => {
  const tree = create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});