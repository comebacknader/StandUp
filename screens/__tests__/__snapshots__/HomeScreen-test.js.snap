// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the HomeScreen component 1`] = `
<View>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "row",
              },
              Object {
                "alignItems": "center",
              },
              Object {
                "padding": 10,
              },
              Object {
                "borderBottomColor": "lightgray",
                "borderBottomWidth": 1,
              },
            ]
          }
        >
          <Image
            source="test-file-stub"
            style={
              Array [
                Object {
                  "height": 60,
                  "width": 60,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
                Object {
                  "fontFamily": "RibeyeMarrow-Regular",
                },
              ]
            }
          >
            Impressions
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "flexDirection": "row",
              },
              Object {
                "alignItems": "center",
              },
              Object {
                "padding": 10,
              },
              Object {
                "borderBottomColor": "lightgray",
                "borderBottomWidth": 1,
              },
            ]
          }
        >
          <Image
            source="test-file-stub"
            style={
              Array [
                Object {
                  "height": 60,
                  "width": 60,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "marginLeft": 20,
                },
                Object {
                  "fontFamily": "RibeyeMarrow-Regular",
                },
              ]
            }
          >
            Bits
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
